test/gtest/ucp/test_ucp_sockaddr.cc

```diff
@@ -1010,7 +957,87 @@ UCS_TEST_SKIP_COND_P(test_ucp_sockaddr, listener_invalid_params,
     ucp_listener_destroy(listener);
 }
 
-UCS_TEST_SKIP_COND_P(test_ucp_sockaddr, compare_cm_and_wireup_configs,
+UCP_INSTANTIATE_ALL_TEST_CASE(test_ucp_sockaddr)
+
+
+class test_ucp_sockaddr_wireup : public test_ucp_sockaddr {
+public:
+    static void
+    get_test_variants(std::vector<ucp_test_variant>& variants,
+                      uint64_t features = UCP_FEATURE_TAG) {
+        /* It is enough to check TAG-only, since we are interested in WIREUP
+         * testing only */
+        get_test_variants_cm_mode(variants, features, CONN_REQ_TAG, "tag");
+    }
+
+protected:
+    static void cmp_cfg_lanes(ucp_ep_config_key_t *key1, ucp_lane_index_t lane1,
+                              ucp_ep_config_key_t *key2, ucp_lane_index_t lane2) {
+        EXPECT_TRUE(((lane1 == UCP_NULL_LANE) && (lane2 == UCP_NULL_LANE)) ||
+                    ((lane1 != UCP_NULL_LANE) && (lane2 != UCP_NULL_LANE) &&
+                     ucp_ep_config_lane_is_peer_match(key1, lane1, key2, lane2)));
+    }
+
+
+    typedef enum {
+        FAIL_WIREUP_MSG_SEND,
+        FAIL_WIREUP_MSG_ADDR_PACK,
+        FAIL_WIREUP_SET_EP_FAILED
+    } fail_wireup_t;
+
+    void connect_and_fail_wireup(entity &e, uint32_t wait_ep_flags,
+                                 fail_wireup_t fail_wireup_type)
+    {
+        start_listener(cb_type());
+
+        scoped_log_handler slh(wrap_errors_logger);
+        client_ep_connect();
+        if (!wait_for_server_ep(false)) {
+            UCS_TEST_SKIP_R("cannot connect to server");
+        }
+
+        if (fail_wireup_type == FAIL_WIREUP_MSG_SEND) {
+            /* Emulate failure of WIREUP MSG sending by setting the AM Bcopy
+             * function which always return EP_TIMEOUT error */
+            for (ucp_lane_index_t lane_idx = 0;
+                 lane_idx < ucp_ep_num_lanes(e.ep()); ++lane_idx) {
+                uct_ep_h uct_ep = e.ep()->uct_eps[lane_idx];
+
+                uct_ep->iface->ops.ep_am_bcopy =
+                        reinterpret_cast<uct_ep_am_bcopy_func_t>(
+                                ucs_empty_function_return_bc_ep_timeout);
+            }
+        }
+
+        while (!(e.ep()->flags & wait_ep_flags)) {
```

Comment:
1. ucs_test_all_flags
2. should use ASYNC_BLOCK?

Reply:
nice catch - fixed